### PR TITLE
Improve framework regexes

### LINKED ISSUE #9286: Analyze and refactor framework regexes
|Wazuh version|Component|
|---|---|
| 4.3.0 | Framework |

Hello team,

We have some regexes in the framework that are too broad. We should analyze all of them and consider if it is possible to refactor them into a safer and simpler form.

Example:

https://github.com/wazuh/wazuh/blob/c432b19247542b02b5153c7ed8fbda4545e2620f/framework/wazuh/core/agent.py#L34

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

Regards,
Víctor

### PR DESCRIPTION
**Pull request by vicferpoy**
|Related issue|
|---|
| closes #9286  |

## Description

Hello team,
This PR improves all the framework regexes in order to make them more specific and prevent high CPU usage. The API ones did not need any change.

### Unit tests
```
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.9.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/vicferpoy/Desktop/Git/wazuh/framework
plugins: cov-2.12.0, asyncio-0.15.1
collected 1589 items                                                                                                                                                                        

wazuh/core/cluster/dapi/tests/test_dapi.py .......................                                                                                                                    [  1%]
wazuh/core/cluster/tests/test_cluster.py ...................                                                                                                                          [  2%]
wazuh/core/cluster/tests/test_common.py .......                                                                                                                                       [  3%]
wazuh/core/cluster/tests/test_control.py .....                                                                                                                                        [  3%]
wazuh/core/cluster/tests/test_local_client.py .                                                                                                                                       [  3%]
wazuh/core/cluster/tests/test_utils.py .......                                                                                                                                        [  3%]
wazuh/core/cluster/tests/test_worker.py .................                                                                                                                             [  4%]
wazuh/core/tests/test_active_response.py ..............                                                                                                                               [  5%]
wazuh/core/tests/test_agent.py ...................................................................................................................................................... [ 15%]
................                                                                                                                                                                      [ 16%]
wazuh/core/tests/test_cdb_list.py ......................................                                                                                                              [ 18%]
wazuh/core/tests/test_common.py .......                                                                                                                                               [ 19%]
wazuh/core/tests/test_configuration.py ....................................                                                                                                           [ 21%]
wazuh/core/tests/test_decoder.py ................                                                                                                                                     [ 22%]
wazuh/core/tests/test_input_validator.py ...                                                                                                                                          [ 22%]
wazuh/core/tests/test_logtest.py ..                                                                                                                                                   [ 22%]
wazuh/core/tests/test_manager.py ...............                                                                                                                                      [ 23%]
wazuh/core/tests/test_mitre.py .............                                                                                                                                          [ 24%]
wazuh/core/tests/test_pyDaemonModule.py .....                                                                                                                                         [ 24%]
wazuh/core/tests/test_results.py ........................................                                                                                                             [ 27%]
wazuh/core/tests/test_rootcheck.py ..........                                                                                                                                         [ 27%]
wazuh/core/tests/test_rule.py ......................                                                                                                                                  [ 29%]
wazuh/core/tests/test_stats.py ....                                                                                                                                                   [ 29%]
wazuh/core/tests/test_syscollector.py ...                                                                                                                                             [ 29%]
wazuh/core/tests/test_utils.py ...................................................................................................................................................... [ 39%]
..............................................................................                                                                                                        [ 44%]
wazuh/core/tests/test_vulnerability.py .                                                                                                                                              [ 44%]
wazuh/core/tests/test_wazuh_queue.py ..................                                                                                                                               [ 45%]
wazuh/core/tests/test_wazuh_socket.py ....................                                                                                                                            [ 46%]
wazuh/core/tests/test_wdb.py ......................                                                                                                                                   [ 47%]
wazuh/rbac/tests/test_auth_context.py ..                                                                                                                                              [ 48%]
wazuh/rbac/tests/test_decorators.py .........................................................................................................                                         [ 54%]
wazuh/rbac/tests/test_default_configuration.py .......................................................                                                                                [ 58%]
wazuh/rbac/tests/test_orm.py ......................................................                                                                                                   [ 61%]
wazuh/rbac/tests/test_preprocessor.py ...........                                                                                                                                     [ 62%]
wazuh/tests/test_active_response.py ............                                                                                                                                      [ 62%]
wazuh/tests/test_agent.py .............................................................................................................                                               [ 69%]
wazuh/tests/test_cdb_list.py .....................................................                                                                                                    [ 73%]
wazuh/tests/test_ciscat.py .................................                                                                                                                          [ 75%]
wazuh/tests/test_cluster.py .......                                                                                                                                                   [ 75%]
wazuh/tests/test_decoder.py ...................................                                                                                                                       [ 77%]
wazuh/tests/test_group.py ........                                                                                                                                                    [ 78%]
wazuh/tests/test_logtest.py ......                                                                                                                                                    [ 78%]
wazuh/tests/test_manager.py ....................................                                                                                                                      [ 81%]
wazuh/tests/test_mitre.py .......                                                                                                                                                     [ 81%]
wazuh/tests/test_rootcheck.py ..................................................                                                                                                      [ 84%]
wazuh/tests/test_rule.py ........................................................                                                                                                     [ 88%]
wazuh/tests/test_sca.py .......                                                                                                                                                       [ 88%]
wazuh/tests/test_security.py ............................................................................                                                                             [ 93%]
wazuh/tests/test_stats.py ................                                                                                                                                            [ 94%]
wazuh/tests/test_syscheck.py .......................                                                                                                                                  [ 95%]
wazuh/tests/test_syscollector.py ............                                                                                                                                         [ 96%]
wazuh/tests/test_task.py ............................                                                                                                                                 [ 98%]
wazuh/tests/test_vulnerability.py ..........................                                                                                                                          [100%]

======================================================================= 1589 passed, 14 warnings in 261.78s (0:04:21) =======================================================================
```

Regards,
Víctor

### CODE REVIEW

File: framework/wazuh/core/configuration.py [hunk 1]
```diff
@@ -401,16 +401,15 @@ def _rootkit_files2json(filepath):
 
     # file_name ! Name ::Link to it
     regex_comment = re.compile(r"^\s*#")
-    regex_check = re.compile(r"^\s*(.+)\s+!\s*(.+)\s*::\s*(.+)")
+    regex_check = re.compile(r"^\s?(.+)\s!\s?(.+)\s?::\s?(.+)")
```
Comment (mcarmona99) [inline framework/wazuh/core/configuration.py:404]: In lines 413-414, strips to each group are made, so we could avoid using \s in this regex:

```suggestion
    regex_check = re.compile(r"^(.+)!(.+)::(.+)")
```


This suggested regex is also passed by regexploit:

```
regexploit-py framework/wazuh/core/configuration.py 
Processed 13 regexes
```

File: framework/wazuh/core/configuration.py [hunk 2]
```diff
@@ -432,8 +431,8 @@ def _rootkit_trojans2json(filepath):
 
     # file_name !string_to_search!Description
     regex_comment = re.compile(r"^\s*#")
-    regex_check = re.compile(r"^\s*(.+)\s+!\s*(.+)\s*!\s*(.+)")
-    regex_binary_check = re.compile(r"^\s*(.+)\s+!\s*(.+)\s*!")
+    regex_check = re.compile(r"^\s?(.+)\s!\s?(.+)\s?!\s?(.+)")
+    regex_binary_check = re.compile(r"^\s?(.+)\s!\s?(.+)\s?!")
```
Comment (mcarmona99) [inline framework/wazuh/core/configuration.py:435]: Same as above

File: framework/wazuh/rbac/orm.py [hunk 3]
```diff
@@ -1354,12 +1357,11 @@ def add_policy(self, name: str, policy: dict, check_default: bool = True):
                     if isinstance(policy['actions'], list) and isinstance(policy['resources'], list) \
                             and isinstance(policy['effect'], str):
                         # Regular expression that prevents the creation of invalid policies
```
Comment (mcarmona99) [inline framework/wazuh/rbac/orm.py:1359]: Remove or change this comment as the regex is not defined here

File: framework/wazuh/rbac/orm.py [hunk 4]
```diff
@@ -1488,12 +1490,11 @@ def update_policy(self, policy_id: int, name: str, policy: dict, check_default:
                     if policy is not None and 'actions' in policy.keys() and \
                             'resources' in policy.keys() and 'effect' in policy.keys():
                         # Regular expression that prevents the creation of invalid policies
```
Comment (mcarmona99) [inline framework/wazuh/rbac/orm.py:1492]: Same as above

File: framework/wazuh/rbac/orm.py [hunk 5]
```diff
@@ -1272,6 +1272,9 @@ class PoliciesManager:
     This class is the manager of the Policies, this class provided
     all the methods needed for the policies administration.
     """
+    def __init__(self):
+        self.action_regex = r'^[a-zA-Z_\-]+:[a-zA-Z_\-]+$'
+        self.resource_regex = r'^[a-zA-Z_\-*]+:[\w_\-*]+:[\w_\-\/.*]+$'
```
Comment (mcarmona99) [inline framework/wazuh/rbac/orm.py:1277]: Wouldn't it be more appropriate if these 2 instance variables were class variables? Their values will always be the same in all instances of the class.

**Review by mcarmona99 — CHANGES_REQUESTED** [review]

File: framework/wazuh/core/configuration.py (hunk 1 above)
Comment (mcarmona99) [inline framework/wazuh/core/configuration.py:404]: In lines 413-414, strips to each group are made, so we could avoid using \s in this regex:

```suggestion
    regex_check = re.compile(r"^(.+)!(.+)::(.+)")
```

**Review by mcarmona99 — CHANGES_REQUESTED** [review]

File: framework/wazuh/core/configuration.py (hunk 1 above)
Comment (mcarmona99) [reply framework/wazuh/core/configuration.py:404]: Also check this in the regex defined in lines 434, 435

**Review by mcarmona99 — APPROVED** [review]
LGTM

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**